### PR TITLE
DOC - Add ``bibtex`` to cite ``Benchopt``

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
This adds ``bibtex`` to cite ``Benchopt`` in

- [x] README
- [x] documentation

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -157,6 +157,25 @@ Available optimization problems
...
+Citing Benchopt
+---------------
+
+If you use ``Benchopt`` in scientific publication, please consider citing the following paper
+
+.. code-block:: bibtex
+
+   @article{benchopt,
+      author = {Thomas Moreau, Mathurin Massias, Alexandre Gramfort, Pierre Ablin,
+                Pierre-Antoine Bannier, Benjamin Charlier, Mathieu Dagréou, Tom Dupré la Tour,
```
Comment (mathurinm) [inline README.rst:169]: bibtex may have difficulty splitting between first and last names in this format, you can use Moreau, Thomas and Massias, Mathurin and ... etc

File: README.rst [hunk 2]
```diff
@@ -157,6 +157,25 @@ Available optimization problems
      - |Build Status ResNetClassif|
 
 
+Citing Benchopt
+---------------
+
+If you use ``Benchopt`` in scientific publication, please consider citing the following paper
```
Comment (mathurinm) [inline README.rst:163]:
```suggestion
If you use ``Benchopt`` in a scientific publication, please cite the following paper
```

**Review by mathurinm — APPROVED** [review]

**Review by agramfort — APPROVED** [review]
LGTM !

**Comment by agramfort** [comment]
thx @Badr-MOUFAD 🙏

**Pull request merged**